flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -117,4 +138,17 @@
 
 		return externalResourceDrivers;
 	}
+
+	/**
+	 * Get the external resource information from environment. Index by the resourceName.
+	 */
+	public static Map<String, Set<? extends ExternalResourceInfo>> getExternalResourceInfo(Map<String, ExternalResourceDriver> externalResourceDrivers, Configuration configuration) {
```

Comment:
It seems for all the callers of this method, the two arguments are both derived from an `Environment`.
Can this utils function just take an `Environment` as argument?

Reply:
I think it would be better to not expose all components to this function.